### PR TITLE
Add Test Builders

### PR DESCRIPTION
**Pull request by seanlaiys**
Added Test Builders for Assessment, Student. Added Typical Data for Assessment, Students and Tutorials. Added tests for Tutorial package.

### CODE REVIEW

File: src/test/java/seedu/address/model/tutorial/TutorialTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        // same object -> returns true
+        assertTrue(T01.equals(T01));
+
+        // null -> returns false
+        assertFalse(T01.equals(null));
+
+        // different type -> returns false
+        assertFalse(T01.equals(5));
+
+        // different person -> returns false
```
Comment (garethkoh) [inline src/test/java/seedu/address/model/tutorial/TutorialTest.java:61]:
```suggestion
        // different tutorial -> returns false
```

File: src/test/java/seedu/address/model/tutorial/VenueTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.junit.jupiter.api.Test;
+
+public class VenueTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Time(null));
+    }
+
+    @Test
+    public void constructor_invalidDay_throwsIllegalArgumentException() {
```
Comment (garethkoh) [inline src/test/java/seedu/address/model/tutorial/VenueTest.java:16]:
```suggestion
    public void constructor_invalidVenue_throwsIllegalArgumentException() {
```

File: src/test/java/seedu/address/testutil/TypicalStudents.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    public static final Student CHARLIE = new StudentBuilder().withName("Charlie Johnson")
+            .withAddress("321, Jurong North Ave 9, #10-123").withEmail("cjohnson@hotmail.com")
+            .withPhone("83214567").withTags("student").withStudentId("e0523614")
+            .withTutorialName("TG01").build();
+    public static final Student DENSON = new StudentBuilder().withName("Denson Loh")
+            .withAddress("321, Ang Mo Kio West Ave 9, #10-101").withEmail("dsnlh@hotmail.com")
+            .withPhone("83214567").withTags("student").withStudentId("e0563214")
+            .withTutorialName("TG01").build();
+
+    // Manually added - Student's details found in {@code StudentTestUtil}
```
Comment (garethkoh) [inline src/test/java/seedu/address/testutil/TypicalStudents.java:44]: im not too clear why this is needed, but if you think it should remain then ok

File: src/test/java/seedu/address/logic/commands/AssessmentTestUtil.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package seedu.address.logic.commands;
+
+public class AssessmentTestUtil {
+    public static final String VALID_ASSESSMENT_NAME_OP1 = "Oral Presentation 1";
+    public static final String VALID_ASSESSMENT_NAME_OP2 = "Oral Presentation 1";
```
[the remark this answers is not in the document]
Comment (seanlaiys) [reply src/test/java/seedu/address/logic/commands/AssessmentTestUtil.java:5]: Thanks will make the changes!

File: src/test/java/seedu/address/model/tutorial/TutorialTest.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import static seedu.address.testutil.TypicalTutorials.TG2;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.TutorialBuilder;
+
+public class TutorialTest {
+
+    @Test
+    public void isSameTutorial() {
```
[the remark this answers is not in the document]
Comment (seanlaiys) [reply src/test/java/seedu/address/model/tutorial/TutorialTest.java:20]: This test case is meant to test the method isSameTutorial() in the Tutorial class, which is the weaker notion of equality between two tutorials, similar to the isSamePerson() method in the Person class. Will explain the difference in the header commment for the Tutorial class!

File: src/test/java/seedu/address/model/tutorial/TutorialTest.java (hunk 1 above)
Comment (seanlaiys) [reply src/test/java/seedu/address/model/tutorial/TutorialTest.java:61]: Thanks! Will make the changes

File: src/test/java/seedu/address/model/tutorial/VenueTest.java (hunk 2 above)
Comment (seanlaiys) [reply src/test/java/seedu/address/model/tutorial/VenueTest.java:16]: Thanks! Will make the change.

File: src/test/java/seedu/address/testutil/TypicalStudents.java (hunk 3 above)
Comment (seanlaiys) [reply src/test/java/seedu/address/testutil/TypicalStudents.java:44]: These are the manually added objects whose fields are valid and will be available in their respective class' Test Util to be used for testing. In future, these Test Util files will also contain invalid fields for testing as well.

**Pull request merged**